### PR TITLE
added readme.md and updated documentation in train

### PR DESCRIPTION
**Pull request by bohnetbd**

### CODE REVIEW

File: examples/nlp_seq/train.py [hunk 1]
```diff
@@ -24,13 +24,12 @@
...
 from absl import flags
 from absl import logging
-from flax import jax_utils
 from flax import nn
 from flax import optim
 import input_pipeline
 import models
 from flax.metrics import tensorboard
-from flax.training import common_utils
+from flax.training import common_utils  # from flax.training import common_utils
```
Comment (marcvanzee) [inline examples/nlp_seq/train.py]: Remove comment?

File: examples/nlp_seq/train.py [hunk 2]
```diff
@@ -95,8 +80,7 @@
 @functools.partial(jax.jit, static_argnums=(1, 2))
 def create_model(key, input_shape, model_kwargs):
   model_def = models.Transformer.partial(train=False, **model_kwargs)
-  _, initial_params = model_def.init_by_shape(key, [(input_shape, jnp.float32)])
-  model = nn.Model(model_def, initial_params)
+  _, model = model_def.create_by_shape(key, [(input_shape, jnp.float32)])
```
Comment (marcvanzee) [inline examples/nlp_seq/train.py]: create_by_shape is deprecated, please use init_by_shape instead.

File: examples/nlp_seq/train.py [hunk 3]
```diff
@@ -109,6 +93,7 @@ def create_optimizer(model, learning_rate):
       weight_decay=FLAGS.weight_decay)
   optimizer = optimizer_def.create(model)
   optimizer = jax_utils.replicate(optimizer)
+  optimizer = optimizer.replicate()
```
Comment (marcvanzee) [inline examples/nlp_seq/train.py]: This is also deprecated, you should instead use jax_utils.replicate(optimizer)

File: examples/nlp_seq/train.py [hunk 4]
```diff
@@ -251,10 +236,7 @@ def loss_fn(model):
 
   step = optimizer.state.step
   lr = learning_rate_fn(step)
-  grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
-  (_, logits), grad = grad_fn(optimizer.target)
-  grad = jax.lax.pmean(grad, 'batch')
-  new_optimizer = optimizer.apply_gradient(grad, learning_rate=lr)
+  new_optimizer, _, logits = optimizer.optimize(loss_fn, learning_rate=lr)
```
Comment (marcvanzee) [inline examples/nlp_seq/train.py]: Also deprecated :)

File: examples/nlp_seq/train.py (hunk 2 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py]: Done. Thanks!

File: examples/nlp_seq/train.py (hunk 3 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py]: Done.

File: examples/nlp_seq/train.py (hunk 4 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py]: changed.

File: examples/nlp_seq/train.py (hunk 1 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py]: Thanks. removed

**Comment by bohnetbd** [comment]
Thanks. I changed the parts in question.

File: examples/nlp_seq/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+Download via command line: 
+
+    curl -# -o ud-treebanks-v2.0.tgz https://lindat.mff.cuni.cz/repository/xmlui/bitstream/handle/11234/1-1976/ud-treebanks-v2.0.tgz
+    tar xzf ud-treebanks-v2.0.tgz 
+
+### Supported setups
+The model should run with other configurations and hardware, but explicitly tested on the following.
+
+| Hardware |  Batch size  | Learning rate | Steps/Second | Accuracy  | TensorBoard |
+|:---:|:---:|:---:|:---:|:---:|:---:|
```
Comment (marcvanzee) [inline examples/nlp_seq/README.md]: Is this indented correctly? It looks a bit strange.

File: examples/nlp_seq/train.py (hunk 4 above)
Comment (marcvanzee) [reply examples/nlp_seq/train.py]: I still see optimizer.optimize(), did something go wrong?

File: examples/nlp_seq/train.py [hunk 6]
```diff
@@ -418,6 +402,12 @@ def main(argv):
           jnp.exp(eval_summary['loss']), a_max=1.0e4)
       logging.info('eval in step: %d, loss: %.4f, accuracy: %.4f', step,
                    eval_summary['loss'], eval_summary['accuracy'])
+
+      if best_dev_score < eval_summary['accuracy']:
+        best_dev_score = eval_summary['accuracy']
+        # save model.
```
Comment (marcvanzee) [inline examples/nlp_seq/train.py]: Should this have 2 spaces indentation less?

File: examples/nlp_seq/README.md (hunk 5 above)
Comment (bohnetbd) [reply examples/nlp_seq/README.md]: |:---:| means center and shows the content better separated.

Comment (marcvanzee) [reply examples/nlp_seq/README.md]: Ahh I didn't know that markdown, nice :)

File: examples/nlp_seq/train.py (hunk 6 above)
Comment (marcvanzee) [reply examples/nlp_seq/train.py]: Please change this to a TODO.

**Pull request merged**